src/decorator.ts

```diff
@@ -70,3 +73,71 @@ export const Paginate = createParamDecorator((_data: unknown, ctx: ExecutionCont
         path,
     }
 })
+
+/* istanbul ignore next */
+function getEntityTree(entity: EntityTarget<any>, tree: Record<string, ''>, parent?: RelationMetadataArgs) {
+    const metadataStorage = getMetadataArgsStorage()
+    const relations = metadataStorage.relations.filter((x) => x.target === entity)
+
+    metadataStorage.columns
+        .filter((x) => x.target === entity)
+        .forEach((column) => {
+            tree[(parent ? `${parent.propertyName}.` : '') + column.propertyName] = ''
+        })
+
+    relations.forEach((relation) => {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const relationEntity = relation.type()
+
+        if (!parent || parent.target !== relationEntity) {
+            getEntityTree(relationEntity, tree, relation)
+        }
+    })
+}
+
+/* istanbul ignore next */
+export const PaginateDocs = (entity: EntityTarget<any>) => {
+    const tree: { [name: string]: any } = {}
+
+    getEntityTree(entity, tree)
+
+    const filterQueries = Object.keys(tree).map((key) => {
```

Comment:
sorry for letting this become dormant.

I think tho we need a better solution here. just printing all the fields of an entity and all the operators will just boat the docs. and as it works on a whitelist basis, it will not be correct for most cases.

would be great if you can come up with a solution which can come up with more precise docs and may not even require a double declaration. maybe the whitelist from the conf can somehow be inferred here.